dnf-behave-tests/features/epoch.feature

```diff
@@ -0,0 +1,48 @@
+Feature: Test upgrading and downgrading to package with different epoch
+
+
+Scenario: Upgrade to RPM with same NVR but different epoch
+  Given I use repository "epoch0"
+    And I successfully execute dnf with args "install dummy"
+    And I use repository "epoch1"
+   When I execute dnf with args "upgrade dummy"
+   Then the exit code is 0
+    And Transaction is following
+        | Action        | Package                |
+        | upgrade       | dummy-1:1.0-1.x86_64   |
+
+
+@xfail
+Scenario: Upgrade to RPM with same NVR but different epoch from RPM with no epoch
+  Given I use repository "epoch"
+    And I successfully execute dnf with args "install dummy"
+    And I use repository "epoch1"
+   When I execute dnf with args "downgrade dummy"
```

Comment:
It says "Upgrade" in Scenario name, but here it does a downgrade...

Also, I've come to a conclusion the `@xfail` tests are really not that useful, since they are always ignored (as in not run) and therefore they rot. It would be great if we could fix the problems instead, as it seems the `@xfail`s are just "never" going to be addressed. Just ranting though, it's still better than nothing :slightly_smiling_face:

Reply:
> It says "Upgrade" in Scenario name, but here it does a downgrade...

Sorry about that, I fixed that. Also, this scenario is actually not failing, so I removed ``@xfail``.

> Also, I've come to a conclusion the `@xfail` tests are really not that useful, since they are always ignored (as in not run) and therefore they rot. It would be great if we could fix the problems instead, as it seems the `@xfail`s are just "never" going to be addressed. Just ranting though, it's still better than nothing slightly_smiling_face

I wrote these tests when investigating bug 1845069. Normally, I would write patch and then these tests wouldn't be tagged ``@xfail``, but since the bug was reassigned to rpm, I don't have control over when it's going to be fixed.